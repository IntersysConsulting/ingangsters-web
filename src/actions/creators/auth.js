import {
  LOGIN_SUCCESS,
  LOGIN_FAIL,
  LOGINADMIN_SUCCESS,
  LOGIN_ADMINFAIL
} from "../../actions/types/auth";
import axios from "axios";
import { API } from "../../config";

export const login = (email, password) => async dispatch => {
  const config = {
    headers: {
      "Content-Type": "application/json"
    }
  };
  const body = JSON.stringify({ email, password });
  try {
    const res = await axios.post(`${API}/user/login`, body, config);
    dispatch({
      type: LOGIN_SUCCESS,
      payload: res.data
    });
  } catch (err) {
    console.log(err);
    dispatch({
      type: LOGIN_FAIL
    });
  }
};

export const adminLogin = (email, password) => async dispatch => {
  const config = {
    headers: {
      "Content-Type": "application/json"
    }
  };
  const body = JSON.stringify({ email, password });
  try {
    const res = await axios.post(`${API}/admin/login`, body, config);
    dispatch({
      type: LOGINADMIN_SUCCESS,
      payload: res.data
    });
  } catch (err) {
    console.log(err);
    dispatch({
      type: LOGIN_ADMINFAIL
    });
  }
};
