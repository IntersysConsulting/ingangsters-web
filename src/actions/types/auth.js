export const LOGIN_SUCCESS = "LOGIN_SUCCESS";
export const LOGIN_FAIL = "LOGIN_FAIL";
export const LOGINADMIN_SUCCESS = "LOGINADMIN_SUCCESS";
export const LOGIN_ADMINFAIL = "LOGIN_ADMINFAIL";
