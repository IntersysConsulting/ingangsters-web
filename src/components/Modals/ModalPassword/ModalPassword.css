.AskPasswordModal {
  width: 270px;
  height: 170px;
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  background-color: white;
  align-content: center;
  border-radius: 5px;
}
.PasswordModalContent {
  padding: 0;
  padding-left: 1%;
}
.PasswordModalLabel {
  text-align: center;
  font-weight: 500;
}
.form-control-password-modal {
  border-color: #000000;
  border-width: 1px;
  height: 30px;
  margin-right: 50%;
  width: 200px;
  border-radius: 5%;
}
.AskPasswordModal .modal-content {
  border: 0;
}

.password-form-modal {
  margin-left: 10%;
}
.modal-open .modal {
  overflow-x: hidden;
  overflow-y: hidden;
}

.AskPasswordModalSendButton {
  background-color: var(--green) !important;
  border-radius: 5px !important;
  border-color: var(--green) !important;
  width: 150px;
  height: 40px;
  display: block;
  margin: auto;
  margin-top: 3%;
}
