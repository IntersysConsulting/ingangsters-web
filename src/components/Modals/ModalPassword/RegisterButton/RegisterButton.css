.RegisterButtonModal {
  width: 100px;
  height: 40px;
  margin-left: 10%;
}
