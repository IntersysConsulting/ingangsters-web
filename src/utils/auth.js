import axios from "axios";
import { API } from "../config";

export const isAuthenticated = () => {
  if (typeof window == "undefined") {
    return false;
  }
  return localStorage.getItem("token") ? true : false;
};

export const isAdmin = async () => {
  const token = localStorage.getItem("token");
  const config = {
    headers: {
      Authorization: `Bearer ${token}`
    }
  };
  try {
    const res = await axios.post(`${API}/admin/check`, {}, config);
    console.log("res:", res.data.isAdmin);
    return res.data.isAdmin;
  } catch (err) {
    console.log(err);
    return false;
  }
};
