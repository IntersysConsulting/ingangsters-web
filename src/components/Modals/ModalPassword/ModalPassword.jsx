import React, { useState } from "react";
import { Button } from "react-bootstrap";
import Modal from "react-bootstrap/Modal";
import RegisterButton from "./RegisterButton/RegisterButton";
import "../../../css/colors.css";
import "./ModalPassword.css";
import { connect } from "react-redux";
import { checkPassword } from "../../../actions/creators/auth";

function ModalPassword({ checkPassword, confirmed_password }) {
  const [values, setValues] = useState({
    password: ""
  });
  const { password } = values;

  const handleChange = name => event => {
    setValues({ ...values, error: false, [name]: event.target.value });
  };
  const [show, setShow] = React.useState(false);
  let handleClose = () => setShow(false);
  let handleShow = () => setShow(true);

  const onSubmit = async event => {
    event.preventDefault();
    checkPassword(password);
  };

  if (confirmed_password) {
    alert("Password Confirmed");
  }

  return (
    <React.Fragment>
      <RegisterButton onClick={handleShow} />
      {confirmed_password !== true ? (
        <Modal show={show} onHide={handleClose} className="AskPasswordModal">
          <Modal.Body className="PasswordModalContent">
            <form>
              <label className="PasswordModalLabel">
                Confirm the register of a New Admin User
              </label>
              <div className="password-form-modal">
                <input
                  onChange={handleChange("password")}
                  type={"password"}
                  placeholder="  Password"
                  className="form-control-password-modal"
                />
              </div>
            </form>
            <Button className="AskPasswordModalSendButton" onClick={onSubmit}>
              Send
            </Button>
          </Modal.Body>
        </Modal>
      ) : (
        <div></div>
      )}
    </React.Fragment>
  );
}
const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticatedl,
  confirmed_password: state.auth.confirmed_password
});

export default connect(
  mapStateToProps,
  { checkPassword }
)(ModalPassword);
