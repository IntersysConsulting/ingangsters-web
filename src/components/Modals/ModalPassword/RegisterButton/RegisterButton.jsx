import React from "react";
import { Button } from "react-bootstrap";
import "./RegisterButton.css";
import "../../../../css/colors.css";

const RegisterButton = props => {
  return (
    <Button
      className="RegisterButtonModal"
      variant="primary"
      onClick={props.onClick}
    >
      Register
    </Button>
  );
};

export default RegisterButton;
