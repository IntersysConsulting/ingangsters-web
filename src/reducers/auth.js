import {
  LOGIN_SUCCESS,
  LOGIN_FAIL,
  LOGINADMIN_SUCCESS,
  LOGIN_ADMINFAIL,
  GET_USER,
  AUTH_ERROR
} from "../actions/types/auth";
import { isAuthenticated } from "../utils/auth";

const initialState = {
  token: localStorage.getItem("token"),
  isAuthenticated: isAuthenticated(),
  loading: true,
  user: null
};

export default function(state = initialState, action) {
  const { type, payload } = action;

  switch (type) {
    case GET_USER:
      return {
        ...state,
        isAuthenticated: true,
        loading: false,
        user: payload
      };
    case LOGIN_SUCCESS:
      localStorage.setItem("token", payload.data.token);
      return {
        ...state,
        token: localStorage.getItem("token"),
        isAuthenticated: true,
        loading: false,
        user: payload.data
      };
    case LOGINADMIN_SUCCESS:
      localStorage.setItem("token", payload.data.token);
      return {
        ...state,
        ...payload,
        token: localStorage.getItem("token"),
        isAuthenticated: true,
        loading: false
      };
    case AUTH_ERROR:
    case LOGIN_FAIL:
    case LOGIN_ADMINFAIL:
      localStorage.removeItem("token");
      return {
        ...state,
        token: null,
        isAuthenticated: false,
        loading: false,
        error: payload
      };
    default:
      return state;
  }
}
