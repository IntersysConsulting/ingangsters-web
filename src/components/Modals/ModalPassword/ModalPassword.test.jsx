import Adapter from "enzyme-adapter-react-16";
import React from "react";
import Enzyme, { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import ModalPassword from "./ModalPassword";
import RegisterButton from "./RegisterButton/RegisterButton";
import { checkPassword } from "../../../actions/creators/auth";

Enzyme.configure({ adapter: new Adapter() });

describe("Confirmation Password Modal", () => {
  const initialState = {
    auth: {
      isAuthenticated: true,
      confirmed_password: false
    }
  };

  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const actions = [checkPassword];
  const store = mockStore(initialState, actions);

  it("Renders correctly", () => {
    const wrapper = renderer.create(<ModalPassword store={store} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  let wrapper = mount(
    <MemoryRouter>
      <ModalPassword store={store} dispatch={null} />
    </MemoryRouter>
  );

  it("Doens't display Modal before RegisterButton Click", () => {
    expect(wrapper.find(".PasswordModalContent")).toHaveLength(0);
  });

  it("Displays Modal after RegisterButton click", () => {
    wrapper.find(RegisterButton).simulate("click");
    expect(wrapper.find(".PasswordModalContent")).toHaveLength(2);
  });

  it("Contains password field and button", () => {
    wrapper.find(RegisterButton).simulate("click");
    expect(wrapper.find(".password-form-modal").length).toEqual(1);
    expect(wrapper.find(".AskPasswordModalSendButton").length).toEqual(2);
  });
});
