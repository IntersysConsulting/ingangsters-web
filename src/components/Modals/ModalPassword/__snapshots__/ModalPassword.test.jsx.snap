// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation Password Modal Renders correctly 1`] = `
<button
  className="RegisterButtonModal btn btn-primary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Register
</button>
`;
